api/envoy/extensions/access_loggers/stats/v3/stats.proto

```diff
@@ -0,0 +1,106 @@
+syntax = "proto3";
+
+package envoy.extensions.access_loggers.stats.v3;
+
+import "envoy/config/core/v3/extension.proto";
+import "envoy/type/matcher/v3/string.proto";
+
+import "google/protobuf/wrappers.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.access_loggers.stats.v3";
+option java_outer_classname = "StatsProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/access_loggers/stats/v3;statsv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Stats logger]
+// [#extension: envoy.access_loggers.stats]
+
+// [#next-free-field: 6]
+message Config {
+  option (xds.annotations.v3.message_status).work_in_progress = true;
+
+  message TagFormatString {
+    string format_string = 1;
+
+    // Valid values for this tag. If empty, any value is allowed. Must match at least 1 matcher.
+    repeated type.matcher.v3.StringMatcher allowed_values = 2;
+  }
+
+  message TagValue {
+    // onoeof these
+    string fixed_string = 1;
+
+    TagFormatString format_string = 2;
+  }
+
+  message Tag {
+    string name = 1;
+
+    TagValue value = 2;
+  }
+
+  message Stat {
+    string name = 1 [(validate.rules).string = {min_len: 1}];
+
+    repeated Tag tags = 2;
+
+    google.protobuf.UInt32Value max_cardinality = 3;
```

Comment:
What this means and how it works?

Reply:
My thought was to somehow count how many metrics have been created for this, and stop after this count. But now that I've started implementing it, I'm not sure how this would be done, especially if metrics expire sometimes. I think I'll remove it for now, and keep things simple.

Reply:
Can we punt on this for now? It can be done in the low-level stats system since we keep track of the maps per scope. But it has to be per-scope, so we need a shared scope across filter instances.